app/actions/websocket/preferences.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import DatabaseManager from '@database/manager';
+import {WebSocketMessage} from '@typings/api/websocket';
+
+export async function handlePreferenceChangedEvent(serverUrl: string, msg: WebSocketMessage): Promise<void> {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    try {
+        const preference = JSON.parse(msg.data.preference) as PreferenceType;
+        const operator = database?.operator;
+        if (operator) {
+            operator.handlePreferences({
+                prepareRecordsOnly: false,
+                preferences: [preference],
+                sync: true,
+            });
+        }
+    } catch (error) {
+        // Do nothing
+    }
+}
+
+export async function handlePreferencesChangedEvent(serverUrl: string, msg: WebSocketMessage): Promise<void> {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    try {
+        const preferences = JSON.parse(msg.data.preferences) as PreferenceType[];
+        const operator = database?.operator;
+        if (operator) {
+            operator.handlePreferences({
+                prepareRecordsOnly: false,
+                preferences,
+                sync: true,
+            });
+        }
+    } catch (error) {
+        // Do nothing
+    }
+}
+
+export async function handlePreferencesDeletedEvent(serverUrl: string, msg: WebSocketMessage): Promise<void> {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    try {
+        const preferences = JSON.parse(msg.data.preferences) as PreferenceType[];
+        const operator = database?.operator;
+        if (operator) {
+            operator.handlePreferences({
+                prepareRecordsOnly: false,
+                preferences,
+                sync: true,
+            });
```

Comment:
this won't work.. if the preference is present, it will in fact get deleted but then re-added as that is the nature of the handler, in this case you should query the preference from the database and then delete it appropriately

Reply:
When looking at the database, I see the value of the preference is what changes.  In the case of a flagged post, it is being set to `true` when flagged, and the value is getting deleted when unflagged. But, the preference record remains in the db. 

From what you are saying, the ask is to remove the record instead of leaving the record with a null value. 

I'll work on that.